test/functional/test_framework/test_node.py

```diff
@@ -158,7 +157,7 @@ def node_encrypt_wallet(self, passphrase):
         self.encryptwallet(passphrase)
         self.wait_until_stopped()
 
-    def add_p2p_connection(self, p2p_conn, **kwargs):
+    def add_p2p_connection(self, p2p_conn, *args, **kwargs):
```

Comment:
nit: I'd rather do the opposite and disallow passing through unnamed args here. Forcing code to be explicit should be safer and ease future review, imo.

 ```py
def add_p2p_connection(self, p2p_conn, *, **kwargs)
```

Reply:
If the goal is to disallow unnamed arguments, this place to do is where the arguments are actually declared (in the peer_connect method). If you make that change here instead of there, you will be introducing an inconsistency between add_p2p_connection and peer_connect. It is better for testnode to just forward arguments on to peernode, instead making unnecessary assumptions about how it works internally and creating unnecessary differences. Forwarding *args and **kwargs is also idiomatic in python while forwarding just **kwargs is more unusual.

Reply:
I think @ryanofsky is right - the right place for this is in `NodeConnCB.peer_connect()`. Can we leave that for a future PR?

Reply:
@jnewbery Sure, future PR sounds good.